ash/src/extensions/khr/pipeline_binary.rs

```diff
@@ -0,0 +1,95 @@
+//! <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_pipeline_binary.html>
+
+use crate::prelude::*;
+use crate::vk;
+use crate::RawPtr as _;
+use alloc::vec::Vec;
+use core::ffi;
+use core::mem;
+
+impl crate::khr::pipeline_binary::Device {
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCreatePipelineBinariesKHR.html>
+    #[inline]
+    #[doc(alias = "vkCreatePipelineBinariesKHR")]
+    pub unsafe fn create_pipeline_binaries(
+        &self,
+        create_info: &vk::PipelineBinaryCreateInfoKHR<'_>,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+        binaries: &mut vk::PipelineBinaryHandlesInfoKHR<'_>,
+    ) -> VkResult<()> {
+        (self.fp.create_pipeline_binaries_khr)(
+            self.handle,
+            create_info,
+            allocation_callbacks.as_raw_ptr(),
+            binaries,
+        )
+        .result()
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkDestroyPipelineBinaryKHR.html>
+    #[inline]
+    #[doc(alias = "vkDestroyPipelineBinaryKHR")]
+    pub unsafe fn destroy_pipeline_binary(
+        &self,
+        pipeline_binary: vk::PipelineBinaryKHR,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_pipeline_binary_khr)(
+            self.handle,
+            pipeline_binary,
+            allocation_callbacks.as_raw_ptr(),
+        )
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetPipelineKeyKHR.html>
+    #[inline]
+    #[doc(alias = "vkGetPipelineKeyKHR")]
+    pub unsafe fn get_pipeline_key(
+        &self,
+        pipeline_create_info: Option<&vk::PipelineCreateInfoKHR<'_>>,
+    ) -> VkResult<vk::PipelineBinaryKeyKHR<'_>> {
+        let mut pipeline_key = mem::MaybeUninit::uninit();
+        (self.fp.get_pipeline_key_khr)(
+            self.handle,
+            pipeline_create_info.as_raw_ptr(),
+            pipeline_key.as_mut_ptr(),
+        )
+        .assume_init_on_success(pipeline_key)
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetPipelineBinaryDataKHR.html>
+    #[inline]
+    #[doc(alias = "vkGetPipelineBinaryDataKHR")]
+    pub unsafe fn get_pipeline_binary_data(
+        &self,
+        info: &vk::PipelineBinaryDataInfoKHR<'_>,
+        pipeline_binary_key: &mut vk::PipelineBinaryKeyKHR<'_>,
+    ) -> VkResult<Vec<u8>> {
+        // TODO: This returns NOT_ENOUGH_SPACE_KHR, not INCOMPLETE (because nothing is written, instead of returning an impartial result)
+        read_into_uninitialized_vector(|count, data| {
+            (self.fp.get_pipeline_binary_data_khr)(
```

Comment:
Not sure what to do here. We could map `NOT_ENOUGH_SPACE_KHR` to `INCOMPLETE` to make `read_into_uninitialized_vector()` try again, though we could also be changing it at some point to return the incomplete but *partial* result to the user, which is invalid here.

If we don't, we can still use `read_into_uninitialized_vector()` to help us call `vkGetPipelineBinaryDataKHR()` twice. On `NOT_ENOUGH_SPACE_KHR` - which is unexpected - that error is bubbled up to the caller (and they wouldn't know how to handle it, since the wrapper is the one that queried the sizes initially).

Reply:
Mapped to `INCOMPLETE` and rewrote the documentation, also to reference the upstream issue (non-public repo).

@Ralith I think this is ready for merging unless you find anything glaring, and I'll follow up and clean up depending on how the upstream discussion pans out.

Reply:
I wrote a new variant of `read_into_uninitialized_binary_vector()` to deal with this.